### PR TITLE
refactor mechanics to support levels feature

### PR DESCRIPTION
**Pull request by samwyness**
- Added splash screen
- Refactored the speed mechanics to adjust speed at the start of the game, then increase gradually until the max speed is reached
- Added support for levels. Each level will increase the max speed for every 'n' distance units
- Redraw the score panel
- Save the users high score during the session.
  - TODO: save highscore across multiple session
  - TODO: show current highscore marking on the road ([see figma](https://www.figma.com/design/JYG1TxlQ0VJlDbF3Mdjny0/Jams?node-id=66-168&t=xTlPXSXC4LNEH0pp-4))

https://github.com/user-attachments/assets/c69432b7-9cb3-4968-bdf6-b6219d45a28d

### CODE REVIEW

File: scripts/scr_print/scr_print.gml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/// Helper functoin function for printing debug messages with multiple args
+function print(){
```
Comment (BenSmith123) [inline scripts/scr_print/scr_print.gml:2]: ah nice, `print` is much easier than `show_debug_message`

gamemakers way of handing arguments is pretty odd lol

File: scripts/Globals/Globals.gml [hunk 2]
```diff
@@ -1,31 +1,16 @@
...
 	// DEBUG
 	global.show_debug = false
 
 	// SETUP
-	
 	global.display_width = 1920
 	global.display_height = 1080
-	
-	
-	// IN GAME
```
Comment (BenSmith123) [inline scripts/Globals/Globals.gml:21]: nice, I wasn't sure how many objects would reference these so made them global when they didn't need to be

File: objects/obj_controller/Draw_64.gml [hunk 3]
```diff
@@ -2,35 +2,42 @@
...
+	draw_text(distance_x+70, distance_y+22, number_format_commas(round(current_distance)) + "M")
 }
 
 // show game over screen
-if global.game_state == GameState.Finished
+if current_game_state == GameState.Finished
 {
 	// blue background
-	draw_sprite_stretched_ext(spr_overlay_blue, 0, 0, 0, room_width, room_height, image_blend, 0.75)
+	draw_sprite_stretched_ext(spr_color_light_blue, 0, 0, 0, room_width, room_height, image_blend, 0.80)
```
Comment (BenSmith123) [inline objects/obj_controller/Draw_64.gml:22]: is this sprite light blue or just white?

File: objects/obj_splash_manager/Draw_64.gml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+draw_set_color(#0F68EF)
+
+var center_x = room_width/2
+
+draw_sprite(spr_logo_medrecruit_blue, 0, center_x, 340)
+
+draw_set_halign(fa_center) // center text
+
+draw_set_font(font_press_start_lg)
+draw_text(center_x, 520, "MEDRACER")
```
Comment (BenSmith123) [inline objects/obj_splash_manager/Draw_64.gml:15]: we should rename the project in setting so the window etc. show MedRacer as the name

**Review by BenSmith123 — APPROVED** [review]
amazing 🚀 🚀

File: objects/obj_controller/Draw_64.gml (hunk 3 above)
Comment (samwyness) [reply objects/obj_controller/Draw_64.gml:22]: Its a light blue from Pulse UI `#F2F7FF`

File: objects/obj_splash_manager/Draw_64.gml (hunk 4 above)
Comment (samwyness) [reply objects/obj_splash_manager/Draw_64.gml:15]: Will do!

File: scripts/scr_print/scr_print.gml (hunk 1 above)
Comment (samwyness) [reply scripts/scr_print/scr_print.gml:2]: Yea, had me stuck for while! Wasnt sure why my int values werent showing in the console output 😅

**Pull request merged**